### ISSUE TITLE
wrapBlockAtRange() logic seems slightly off

### ISSUE THREAD

**Issue by tyler-johnson** [ianstormtaylor/slate#340]
It seems to me that the way the `wrapBlockAtRange()` transform works is a little off of what would be expected, however I might be totally wrong.

The logic is currently attempting to find all the sibling block nodes in the selection that are closest to root and wraps those. The basis of this logic is what is expected, but I believe there is a little bug that stems from the use of `getBlocksAtRange()` and `getDepth()`.

Consider the following document that has two `paragraph` nodes inside two `list-item` nodes:

```yaml
document:
  type: document
  nodes:
    - type: list-item
      nodes:
        - type: paragraph
    - type: list-item
      nodes:
        - type: paragraph
```

I want to wrap both `list-item` nodes in a `bulleted-list` node, so I select everything in the document and run `transform.wrapBlock('bulleted-list')`. What I am expecting to get back is this:

```yaml
document:
  type: document
  nodes:
    - type: bulleted-list
      nodes:
        - type: list-item
          nodes:
            - type: paragraph
        - type: list-item
          nodes:
            - type: paragraph
```

However what I end up getting is this:

```yaml
document:
  type: document
  nodes:
    - type: list-item
      nodes:
        - type: bulleted-list
          nodes:
            - type: paragraph
            - type: paragraph
    - type: list-item
```

This is happening because `getBlocksAtRange()` return the bottom-most nodes and the use of `getDepth()` to determine the best siblings. Since both of the `paragraph` nodes in the example are at the same depth, the logic treats them as siblings, even though they are not siblings.

I believe the correct logic would look for the closest shared parent of the bottom-most nodes and wrap all the children nodes to that parent that fall within the selection.

**Comment by ianstormtaylor** [comment]
Wow, good find! The proper result should definitely be your expected YAML. If you're down to look into this, I'd love a PR with the fix! If not, a PR with a failing test case would be awesome too! Thanks man

**Comment by tyler-johnson** [comment]
fixed with #343
